Makefile

```diff
@@ -42,6 +42,14 @@ run-flask: ## Run flask
 run-flask-with-docker: ## Run flask
 	./scripts/run_locally_with_docker.sh api-local
 
+.PHONY: run-local-tests-with-docker
+run-local-tests-with-docker:
+	docker build -f docker/Dockerfile --target concourse_tests -t notifications-api .
```

Comment:
This will force a slow rebuild every time the command is run. In Concourse that's probably fine; locally it's potentially annoying.

However I don't think we should be running this locally, so it's also probably fine to ignore.

Reply:
you do not need to run make...if building locally. This was done to simplify the tests locally.

Reply:
So ironically, I don't think we should ever be running `make run-local-tests-with-docker` on Concourse (Sam, you referenced an expectation that it would be run in concourse which I don't think should be the plan).

The reason we won't be running this command in Concourse is because Concourse will enter the docker image using bash, and then will then run whichever commands are specified in the concourse job. The image will already be built by this point.

See https://github.com/alphagov/notifications-aws/blob/main/concourse/templates/api.yml.j2#L64-L66 for how we do this already for the API.

So `run-local-tests-with-docker` would only be useful if developers want/need to run that command from their local machines.


I think you have 2 options here.

1. Remove the `run-local-tests-with-docker` command entirely. If we just want to focus on CONCOURSE running the unit tests in a docker image, then we don't need this make command because it would only be used by devs on their local machines. (Yes, it would be nice to put this functionality in place but could always come in a separate PR or later)

2. Continue with the approach of letting devs run the unit tests in a docker image using Make, however we might want to consider a few tweaks e.g.

- if you run the command twice, it passes the first time and then fails the second
- if you run the command once, and then run `make run-flask-with-docker` that has also broken
- I'm not sure if `./scripts/run_locally_with_docker.sh api-local` is the right thing to run. That will leave the app running in the container after the tests have finished

Reply:
Hm, interesting.

I got that expectation from the Dockerfile step being called `concourse_tests`, I suppose, and since this new setup probably makes testing a bit of a painfully slow loop (since it'll rebuild the image every time).

I think focusing this on just what's needed for concourse for now makes sense 👍